scalardb-test/src/main/java/kelpie/scalardb/transfer/WriteSkewChecker.java

```diff
@@ -0,0 +1,137 @@
+import com.scalar.kelpie.config.Config;
+import com.scalar.kelpie.exception.PostProcessException;
+import com.scalar.kelpie.modules.PostProcessor;
+import java.util.List;
+import javax.json.JsonArray;
+import javax.json.JsonObject;
+import kelpie.scalardb.Common;
+
+public class WriteSkewChecker extends PostProcessor {
+  private final boolean isSerializable;
+  private final boolean isExtraRead;
+
+  public WriteSkewChecker(Config config) {
+    super(config);
+    this.isSerializable = config.getUserBoolean("test_config", "is_serializable", false);
+    this.isExtraRead = config.getUserBoolean("test_config", "is_extra_read", false);
+  }
+
+  @Override
+  public void execute() {
+    boolean isConsistent = false;
+    try {
+      logInfo("reading latest records ...");
+      List<Result> results = TransferCommon.readRecordsWithRetry(config);
+
+      int numUpdates = getNumOfUpdatesFromCoordinator(config);
+
+      isConsistent = isConsistent(results, numUpdates);
+    } catch (RuntimeException e) {
+      throw new PostProcessException("Failed to read records", e);
+    }
+
+    if (!isConsistent) {
+      throw new PostProcessException("Inconsistency happened!");
+    }
+  }
+
+  @Override
+  public void close() {}
+
+  private int getNumOfUpdatesFromCoordinator(Config config) {
+    Coordinator coordinator = new Coordinator(Common.getStorage(config));
+    JsonObject unknownTransactions = getPreviousState().getJsonObject("unknown_transaction");
+    if (unknownTransactions == null) {
+      // for --only-post
+      return 0;
+    }
+
+    int numUpdates = 0;
+    for (String txId : unknownTransactions.keySet()) {
+      logInfo("checking the status of " + txId);
+      boolean isCommitted = Common.isCommitted(coordinator, txId);
+      if (isCommitted) {
+        JsonArray ids = unknownTransactions.getJsonArray(txId);
+        int fromId = ids.getInt(0);
+        int fromType = ids.getInt(1);
+        int toId = ids.getInt(2);
+        int toType = ids.getInt(3);
+        logInfo(
+            "id: "
+                + txId
+                + " from: "
+                + fromId
+                + ","
+                + fromType
+                + " to: "
+                + toId
+                + ","
+                + toType
+                + " succeeded, not failed");
+        if (isSerializable && !isExtraRead) {
+          if (fromId != toId) {
+            numUpdates += TransferCommon.NUM_TYPES + 1;
+          } else {
+            numUpdates += TransferCommon.NUM_TYPES;
+          }
+        } else {
+          numUpdates += 2;
+        }
+      }
+    }
+
+    return numUpdates;
+  }
+
+  private boolean isConsistent(List<Result> results, int addedNumUpdates) {
+    int totalVersion = TransferCommon.getActualTotalVersion(results);
+    int totalBalance = TransferCommon.getActualTotalBalance(results);
+
+    int numUpdates = 0;
+    try {
+      numUpdates = getPreviousState().getInt("num_updates");
```

Comment:
I think it should be something like this instead of catching NPE.
```
if (!getPreviousState().isNull("num_updates")) {
  numUpdates = getPreviousState().getInt("num_updates");
}
```

Reply:
Thank you for the suggestion! I've modified them.